Docs for Communities (#1526)
matrix-org/matrix.org: content/docs/communities/moderation/_index.md

+++
title = "Community Moderation"
weight = 100
[extra]
emoji = "🛡️"
tile = "I want to deploy moderation tools"
last_updated = "2022-10-27T08:00:00Z"
+++

## Power levels

Matrix has a basic system of roles, sometimes also called Power Levels. It is a
number usually going from 0 to 100. Advanced users can decide to use a different
range, but we will not cover it here for the sake of simplicity.

By default, there are three roles in Matrix:

| Power Level | Role          |
|-------------|---------------|
| 0           | User          |
| 50          | Moderator     |
| 100         | Administrator |

By default:

- **Users** can send messages, media, reactions and redact their own messages in

>   a room.

Reviewer: Is redacting your own messages something that can be configured away? This makes it sounds like that a bit.

Author: It's not, but I wanted to give an overview of the user abilities. Do you have better phrasing in mind?

Reviewer: Could be a separate statement righr before or after the listing that everybody can redact their own messages?

- **Moderators** can also change the room name, address, topic, remove users

>   from the room (temporarily or permanently), redact other's messages, send a

Reviewer: ```suggestion
  from the room (temporarily or permanently), redact other's messages and send a
```

>   noisy notification to everyone in the room

Reviewer: ```suggestion
  notification to everyone in the room at once using `@room`
```

- **Administrators** can also change the history visibility (whether people can
  see messages from before they joined or not), enable encryption in the room,

>   remove entire servers from the room, and promote others as Moderators or
>   Administrators

Reviewer: ```suggestion
  remove entire servers from the room, and promote others to Moderator or
  Administrator
```

Most people in your community are just going to be regular users. Usually
Moderators are appointed to handle the ad-hoc moderation issues. Note that this
can be an issue if you want to preserve the anonymity of your Trust & Safety

> team for their own security. The moderation bot mjolnir brings a good answer to

Reviewer: This sounds quite alarming, and no extra context is given about what this loss of anonymity actually means.

Author: Well it is alarming because it is :)

If the moderation team is not anonymous, they can be doxxed, harassed, and for more extreme cases even assaulted. Retaliation can happen and is a very real threat. I'm going to look for resources to provide more context.

this.

If you're taking over the community of someone who was very technically versed
into Matrix and strongly opinionated on power levels, the roles may not be the
same. In that case, we recommend you to email

> [support@matrix.org](mailto:support@matrix.org) for help.

Reviewer: Do the people that read those emails know you're putting this on the site? 😄

Author: Yes!

## Your community's security guard

> The tools built-in Matrix are good for small group moderation, but they are

Reviewer: ```suggestion
The tools built into Matrix are good for small group moderation, but they are
```

limited for now: removing a user (temporarily or permanently) is only possible
at the room level, and banning an entire server from a room is not exposed in
the UI.

Fortunately, a tool exists to step up your moderation game: a security guard

> robot you add to all of your public rooms, who can enforce your moderation

Reviewer: ```suggestion
robot you add to all of your public rooms that can enforce your moderation
```

decisions on all the rooms at once.

And that's really what mjolnir is: a security guard for your rooms. Because of

> the very decentralised nature of Matrix, you need to add the robot to guard the

Reviewer: ```suggestion
the decentralised nature of Matrix, you need to add the robot to guard the
```

> door of each of your rooms, and you need to grant it the Administrator role so

Reviewer: This is kinda bad, really (because you can't demote it again)..

Author: That's a very good point. In practice it depends:
- If mjolnir runs with an account you created, you can use the credentials to demote it
- If mjolnir runs with an account from a homeserver you have admin access to, you can use the admin API to demote it
- If you run a mjolnir instance with an account you don't have the credentials of, from a homeserver you don't have (indirect) admin access to, then indeed you can't demote it. This is going to be an issue when mjolnir for all is released for example.

The alternative is to update every room to lower several of the permissions from 100 to e.g. 99. It's quite cumbersome and error-prone. Worth discussing internally

Reviewer: IIRC for Ruma I lowered a bunch of things to 50, but maybe it's not a good idea in general 😅

if you have the technical knowledge to do it. We're aware this situation is
suboptimal and are working on a service so people get can mjolnir bots much more
easily.

### Setting it up in your rooms

Once you have performed the technical set-up of mjolnir, you need to:
1. Make sure your moderation staff is in the control room of your community
1. Invite mjolnir the security guard bot in all your rooms
1. Make the mjolnir bot Administrator in all your rooms (so it can enforce
   moderation decisions).

To do so you need to invite the moderation bot in each room. You can do so
either by opening the right panel, opening the members list and clicking on
Invite, or you can type the following message in the message bar at the bottom:

```
/invite @yourMjolnirBot:example.com
```

TODO screenshot of the invite

The bot will join, and you can then promote it Administrator by finding it in
the list of members and changing its role, or you can type the following message
in the message bar at the bottom:

```
/op @yourMjolnirBot:example.com 100
```

TODO screenshot of oping mjolnir

## Getting moderation reports

> - You need your own instance while Aristotle isn't merged in the spec and

Reviewer: Aristotle?

Author: [Nickname of MSC3215](https://github.com/matrix-org/matrix-spec-proposals/pull/3215), those are my notes for when I'll finish the moderation section
